### PR TITLE
chore: Cleanup logic in 'GetOrCreateGitOpsEngine*' (#155)

### PR DESCRIPTION
**Pull request by keithchong**
Signed-off-by: Keith Chong <kykchong@redhat.com>

#### Description:
- Refactor and simplify the logic.
- Two changes related to scope (The use of the curly braces), but it looks like it isn't necessary.  Please confirm.
- Also, I didn't change existing or add any new tests.  They all ran fine.  The various scenarios were not possible anyway.

### CODE REVIEW

**Comment by keithchong** [comment]
```
make test-backend-shared
cd ....../managed-gitops/backend-shared && make test
go fmt ./...
go vet ./...
go test -p=1 -race -count=1 -coverprofile=coverage.out `go list ./...`
?   	github.com/redhat-appstudio/managed-gitops/backend-shared/apis/managed-gitops/v1alpha1	[no test files]
?   	github.com/redhat-appstudio/managed-gitops/backend-shared/apis/managed-gitops/v1alpha1/mocks	[no test files]
?   	github.com/redhat-appstudio/managed-gitops/backend-shared/apis/managed-gitops/v1alpha1/mocks/structs	[no test files]
ok  	github.com/redhat-appstudio/managed-gitops/backend-shared/db	10.384s	coverage: 50.4% of statements
ok  	github.com/redhat-appstudio/managed-gitops/backend-shared/db/util	1.347s	coverage: 55.2% of statements
ok  	github.com/redhat-appstudio/managed-gitops/backend-shared/hack	0.196s	coverage: [no statements]
?   	github.com/redhat-appstudio/managed-gitops/backend-shared/hack/db-schema-sync-check	[no test files]
ok  	github.com/redhat-appstudio/managed-gitops/backend-shared/util	14.104s	coverage: 42.4% of statements
ok  	github.com/redhat-appstudio/managed-gitops/backend-shared/util/argocd	0.499s	coverage: 77.8% of statements
?   	github.com/redhat-appstudio/managed-gitops/backend-shared/util/fauxargocd	[no test files]
?   	github.com/redhat-appstudio/managed-gitops/backend-shared/util/gitopserrors	[no test files]
ok  	github.com/redhat-appstudio/managed-gitops/backend-shared/util/operations	1.108s	coverage: 53.3% of statements
?   	github.com/redhat-appstudio/managed-gitops/backend-shared/util/tests	[no test files]
```

File: backend-shared/db/util/utils.go [hunk 1]
```diff
@@ -194,33 +185,20 @@ func GetOrCreateGitopsEngineInstanceByInstanceNamespaceUID(ctx context.Context,
...
 
-		if err := dbq.GetGitopsEngineInstanceById(ctx, gitopsEngineInstance); err != nil {
-			if !db.IsResultNotFoundError(err) {
-				return nil, false, nil, err
-			}
-
-			log.V(util.LogLevel_Warn).Error(nil,
-				"GetOrCreateGitopsEngineInstanceByInstanceNamespaceUID found a resource mapping, but no engine instance.")
+		var getEngineInstanceErr error
+		if getEngineInstanceErr = dbq.GetGitopsEngineInstanceById(ctx, gitopsEngineInstance); getEngineInstanceErr == nil {
```
Comment (chetan-rns) [inline backend-shared/db/util/utils.go:195]: I prefer to handle the error first and then move forward. Both of them are logically correct, though. :+1:

**Review by chetan-rns — APPROVED** [review]
Thanks, @keithchong 

/lgtm

**Comment by keithchong** [comment]
Hi @jgwest , your suggestions LGTM.

**Review by jgwest — APPROVED** [review]
Thanks @keithchong, and thanks @chetan-rns for reviewing!

**Pull request merged**